### PR TITLE
Feature/chat locker 26 Custom Messages

### PR DESCRIPTION
**Pull request by berkich**

### CODE REVIEW

File: server/methods/setStatusMessage.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	setStatusMessage: (status, message) ->
+		userId = Meteor.userId()
+		if not userId
+			throw new Meteor.Error('invalid-user', "[methods] toogleFavorite -> Invalid user")
+
+		console.log '[methods] setStatusMessage -> '.green, 'userId:', Meteor.userId(), 'arguments:', arguments
+
+		user = Meteor.user()
+		setField = { statusMessages:user.profile.statusMessages}
+		setField.statusMessages[status] = message
```
Comment (rwakida) [inline server/methods/setStatusMessage.coffee]: errors out when user.profile.statusMessages is undefined.  e.g. when no status message has ever been set.

File: server/methods/setStatusMessage.coffee [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+		result = Meteor.users.update userId, {$set: { "profile": setField}}
```
Comment (rwakida) [inline server/methods/setStatusMessage.coffee]: We're overwriting the other "profile" fields (e.g location, access, etc).  try setting the profile.statusMessages field directly.  i.e.

``` javascript
statusMessages = user?.profile?.statusMessages || {}
statusMessages[status] = message
Meteor.users.update userId, {$set : {"profile.statusMessages" : statusMessages}}
```

Comment (berkich) [inline server/methods/setStatusMessage.coffee]: Thanks Reid, making updates

**Comment by bbrockman** [comment]
The functionality works fine when clicking on Online or Away, but nothing happens when I click on busy.  No error or debug logged.

**Comment by rwakida** [comment]
I see it in the private group/ channel members list area when I view the user's profile.  Does it appear anywhere else?

**Comment by rwakida** [comment]
:+1: 

Only fix is to remove colon from SideNav hover if custom status message is empty.  e.g. "online:" to "online"

**Comment by rwakida** [comment]
Not part of the issue's acceptance criteria, but it would be nice to see the status message in members list when ALL members are visible :smile:

**Pull request merged**